common/cuda_hip/matrix/bccoo_kernels.hpp.inc

```diff
@@ -443,7 +380,9 @@ __device__ void absolute_kernel(
     const IndexType start_in_blk = threadIdx.y * subwarp_size + threadIdx.x;
     const IndexType jump_in_blk = blockDim.y * subwarp_size;
 
-    offsets_data_res[0] = 0;
+    if (start_blk == 0) {
+        offsets_data_res[0] = 0;
+    }
```

Comment:
```
if (start_in_blk == 0) {
offsets_data_res[blk + 1] =
            blk_idxs_res.shf_val +
            block_size_local * sizeof(remove_complex<ValueType>);
}
```
also for the offsets_data_res[blk+1]
Because start_blk is from blockIdx.x, the threads in the same block will write the data to memory together.

Reply:
Done!!